sklearn/utils/fixes.py

```diff
@@ -393,3 +393,26 @@ def array_equal(a1, a2):
         return bool(np.asarray(a1 == a2).all())
 else:
     from numpy import array_equal
+
+if 'axis' not in signature(np.linalg.norm).parameters:
+
+    def norm(X, ord=None, axis=None):
+        """
+        Handles the axis parameter for the norm function in old versions of numpy.
+        """
+
+        if axis is None or X.ndim == 1:
+            result = np.linalg.norm(X, ord=ord)
+            return result
+
+        if axis == 0:
```

Comment:
Should we implement the more general case for a nd-array with integer axis using `rollaxis` to move `axis` to the last slot, then reshape so that it's 2d? Alternatively, we should explicitly raise `NotImplementedError` for `axis not in (0, 1)`